### ISSUE TITLE
in some cases the version sorting is not working properly

### ISSUE THREAD

**Issue by dvic** [benlime/vscode-hex-pm-intellisense#18]
Currently, the sorting is done textually it seems:

![image](https://user-images.githubusercontent.com/1214337/217511852-b6db8e71-7ee0-44b8-b558-096e4176fafd.png)

We can improve this by parsing the version according to https://hexdocs.pm/elixir/1.14/Version.html and sort on the individual integers.

**Comment by benvp** [comment]
Can you point me to the package you are trying to complete here? I'm a bit surprise because this should handle the sorting:

https://github.com/benvp/vscode-hex-pm-intellisense/blob/master/src/provide.ts#L93

**Comment by dvic** [comment]
Oh, that's weird, it was `ex_check`

**Comment by dvic** [comment]
It looks like the only way for this to fail is if the `semver` package has a bug? But that seems highly unlikely 😅 `semver.compare("0.9.0", "0.15.0")` should give `-1`

**Comment by benvp** [comment]
Are you using the latest version of the extension? Tried it with that library and I get the following result

<img width="621" alt="CleanShot 2023-02-08 at 12 20 59@2x" src="https://user-images.githubusercontent.com/981344/217515975-a34a2d81-76f4-4e6d-ac67-d2995cd541c4.png">

Any other steps to reproduce it?

**Comment by dvic** [comment]
Yes, I just installed the extension. But I think I found the way to reproduce it:

when you autocomplete inside an empty string, like in your screenshot, it sorts properly, but when you autocomplete inside `"~> 0.15.0"` it does not sort it properly, when I add a space after `"~> 0.15.0"` and then autocomplete, it sorts it correctly again.
